gc/collectors/amis.py

```diff
@@ -44,29 +54,46 @@ def _get_amis(self) -> list[ImageTypeDef]:
 
     def _find_expired_amis(self, amis: list[ImageTypeDef]) -> list[ImageTypeDef]:
         expired_amis = []
-        ami_groups = defaultdict(list)
+        ami_groups: dict[str, dict[str, list[ImageTypeDef]]] = defaultdict(
+            lambda: defaultdict(list)
+        )
 
         # Group AMIs by "image-name" tag
         for ami in amis:
             img_tags = ami["Tags"]
             if img_tags:
+                image_name: Optional[str] = None
+                branch_name: Optional[str] = None
                 for tag in img_tags:
                     if tag["Key"] == "image-name":
-                        img_name = tag["Value"]
-                        ami_groups[img_name].append(ami)
+                        image_name = tag["Value"]
+                    if tag["Key"] == "branch-name":
+                        branch_name = tag["Value"]
+                if image_name:
+                    if branch_name:
+                        ami_groups[branch_name][image_name].append(ami)
                         break
+                    else:
+                        expired_amis.append(ami)
 
         # Sort AMIs by creation date.
         # If image is currently supported, keep only the newest AMI. Expire the rest.
         # If image is not currently supported, expire all AMIs.
-        for img_name, amis in ami_groups.items():
-            amis = sorted(
-                amis, key=lambda x: parser.parse(x["CreationDate"]), reverse=True
-            )
-            if img_name in self.current_images:
-                expired_amis.extend(amis[1:])
-            else:
-                expired_amis.extend(amis)
+        for branch_name, images in ami_groups.items():
+            if branch_name == getenv("BRANCH_NAME"):
```

Comment:
```suggestion
            if branch_name == DEFAULT_BRANCH_NAME
```

can we switch this from using an environment variable to a constant? the default branch name will likely never change. I worry that getting the value from an environment variable could cause issues down the line if someone inadvertently changes or removes the environment variable.

I also think it's probably a good idea to put a check in the `main.py` entrypoint to ensure that the script can only ever be executed from the `main` branch (e.g. `os.getenv("GITHUB_REF") == "main"`). This will avoid accidental issues caused by someone triggering this workflow from the non-default branch.